### ISSUE TITLE
TypeError (can't cast RGeo::Geographic::SphericalPolygonImpl) on heroku

### ISSUE THREAD

**Issue by ather-wal** [rgeo/activerecord-postgis-adapter#278]
Im having a problem while saving circle data to my db. Here is the code, in my before_create callback
`  factory = RGeo::Geographic.spherical_factory(srid: 4326, buffer_resolution: 8)
    data = factory.point(self.longitude, self.latitude).buffer(radius)
    self.circle = data`

I save the data to circle which is of type geometry. It works fine on my local machine. But on heroku I get this error

`TypeError (can't cast RGeo::Geographic::SphericalPolygonImpl)`
Any help?

**Comment by gagalago** [comment]
do you find a way to fix that?

**Comment by gagalago** [comment]
I find the problem. I didn't set in the configuration of the database the adapter as `postgis`. if you use the database throught an url you have to change the protocol part to postgis 
```
adapter: postgis
url: <%= ENV["DATABASE_URL"].sub(/^postgres/, "postgis") %>
```

**Comment by senny** [comment]
We're observing the same error behavior. `postgis` adapter is configured correctly (I can see it in the backtrace):
https://github.com/rgeo/activerecord-postgis-adapter/blob/744bed7e276135be83ebfb6c498273766f918850/lib/active_record/connection_adapters/postgis_adapter.rb#L99-L107

We end up in the `else` branch instead of the RGeo serialization. It also doesn't seem to happen all the time.

**Comment by mjy** [comment]
@keithdoggett Perhaps related to issues closed in 6.0.1 I think?

**Comment by keithdoggett** [comment]
It seems like it's related to the `intialize_type_map` issue, so I'll close it. 

If it's still happening let me know and I can reopen it.
